src/android/CameraLauncher.java

```diff
@@ -690,22 +699,80 @@ private String getExtensionForEncodingType() {
 
 
     /**
-     * Applies all needed transformation to the image received from the gallery.
+     * Applies all needed transformation to the image received from the gallery. Currently limited to three media files, but can be easily modified.
      *
      * @param destType In which form should we return the image
      * @param intent   An Intent, which can return result data to the caller (various data can be attached to Intent "extras").
      */
-    private void processResultFromGallery(int destType, Intent intent) {
-        Uri uri = intent.getData();
-        if (uri == null) {
+    private void processResultFromGallery(int destType, Intent intent) throws JSONException {
+        List<Uri> uriList = new ArrayList<>();
+
+        if (intent.getClipData() == null) {
+            uriList.add(intent.getData());
+        } else {
+            int counter = intent.getClipData().getItemCount();
+            int index = 0;
+            while (index != counter) {
+                uriList.add(intent.getClipData().getItemAt(index).getUri());
+                index++;
+                if (index == 3) // Number "3" limits the selected content from gallery to 3. First three media files are processed.
+                {
+                    break;
+                }
+            }
+        }
+
+        if (uriList.size() == 0) {
             if (croppedUri != null) {
-                uri = croppedUri;
+                uriList.add(croppedUri);
             } else {
                 this.failPicture("null data from photo library");
                 return;
             }
         }
 
+        int pictureCount = uriList.size();
+        JSONArray selectedContent = new JSONArray();
+        int index = 0;
+        if (destType == DATA_URL) {
+            while (index != pictureCount) {
+                String base64Picture = processSingleUri(destType, uriList.get(index));
+                if (base64Picture == null) {
+                    this.failPicture("Error retrieving result.");
+                } else {
+                    selectedContent.put(base64Picture);
```

Comment:
For destType DATA_URL allow only single image selection and return a String to the callback instead of JSONArray due to the stackoverflow error (which prob happens due to the base64 strings being too large for the JSONArray to hold).

Reply:
Done